runtest.py

```diff
@@ -644,41 +619,37 @@ def find_e2e_tests(directory):
         # sys.stderr.write("to:%s\n"%tp)
         for path in glob.glob(tp):
             if os.path.isdir(path):
-                if path.startswith(('SCons', 'testing')) and not e2e_only:
-                    for p in find_unit_tests(path):
-                        unittests.append(p)
-                elif path.startswith('test') and not unit_only:
-                    for p in find_e2e_tests(path):
-                        endtests.append(p)
+                if path.startswith(('SCons', 'testing')):
+                    unittests.extend(find_unit_tests(path))
+                elif path.startswith('test'):
+                    endtests.extend(find_e2e_tests(path))
             else:
-                if path.endswith("Tests.py") and not e2e_only:
+                if path.endswith("Tests.py"):
                     unittests.append(path)
-                else:
-                    if not unit_only:
-                        endtests.append(path)
-
-    tests.extend(unittests)
-    tests.extend(endtests)
-    tests.sort()
+                elif path.endswith(".py"):
+                    endtests.append(path)
+    tests = unittests + endtests
+
+# Remove exclusions:
+if e2e_only:
+    tests = [t for t in tests if not t.endswith("Tests.py")]
+if unit_only:
+    tests = [t for t in tests if t.endswith("Tests.py")]
```

Comment:
Not sure this is reliable way to do this, we don't preclude e2e tests ending in Tests.py

Reply:
it's stated somewhere that if it ends in `Tests.py` it's a unit test. If that's not the case I'll have to find and root out such wording.